### PR TITLE
List All Members with pattern=

### LINKED ISSUE #810: Enable use of a pattern to restrict what member names are returned from listing of PDS members
Currently when listing members from a PDS it is not possible to use a search parameter to restrict the returned member names to only the names that match the given pattern.  

This feature is supported by z/OSMF. have a look here : https://www.ibm.com/support/knowledgecenter/SSLTBW_2.4.0/com.ibm.zos.v2r4.izua700/IZUHPINFO_API_GetListDataSetMembers.htm

It is very an important feature to implement into zowe-cli, because lots of enterprise organizations do have PDS's with large number of members(>200.000),
and its very slow to pull all those member names all the time.  Furthermore we experience other problems when listing PDS with large number of members.

### PR DESCRIPTION
**Pull request by dkelosky**
PR to fix #810.  E.g.:

```
$ zowe files list am "sys1.maclib" --pattern "ikj*" --max 4
IKJADFMT
IKJCAFPL
IKJCNCCB
IKJCNMCB
```

### CODE REVIEW

**Comment by martindamp** [comment]
Thankyou for the fast execution @dkelosky . 
It looks good :-)

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,8 +3,8 @@
 All notable changes to the Zowe CLI package will be documented in this file.
 
 ## Recent Changes
-- Enhancement: Added options like `--lrecl, --recfm` to zos-files create options. [788](https://github.com/zowe/zowe-cli/issues/788)
-***
+
+- Added `--pattern` option to `zowe files list am`
```
Comment (zFernand0) [inline CHANGELOG.md:7]: Please add a label (e.g. `Enhancement:`) and the issue number to the changelog.
Just trying to follow what the Zowe Docs Squad suggested 😋

**Review by zFernand0 — APPROVED** [review]
Works great! 👍 
Thanks for implementing

**Review by awharn — APPROVED** [review]
Looks great. As soon as Fernando's comments are addressed, it should be good to merge. 👍

File: CHANGELOG.md (hunk 1 above)
Comment (dkelosky) [reply CHANGELOG.md:7]: Can you link the doc squad comments? I don't think this makes sense and conflicts with our guidelines in our repo.

**Pull request merged**